osh/builtin_printf.py

```diff
@@ -252,14 +314,31 @@ def Run(self, cmd_val):
                 s = '%x' % d
               elif typ == 'X':
                 s = '%X' % d
+            elif part.type.id == Id.Format_Time:
+              # set timezone
```

Comment:
Can you explain this part in a high level comment?  I don't know these APIs very well and eventually they will have to be translated to C.

(Actually if you prefer to write in C, then `native/libc.c` has related functions and is fairly easy to modify.)

Does `strftime` read `os.environ['TZ']`?  I'm not sure how it works exactly.

-----

OK I looked over the docs a bit, I guess `time.tzset()` reads `os.environ['TZ']`.  That's kind of confusing.  Please add a comment about that.

Reply:
dcf99bd Thank you! I added code comments.